azerothcore/azerothcore-wotlk
# feat(Core/Player): Implement player specific settings

## Changes Proposed:
-  Implement a feature to allows saving specific character settings, that can expand certain server features and can also be used by modules to store character specific options.
-  Implement a command to allow disabling/enabling battleground/arena queues, as well as autobroadcast messages.
- Implement two new world configs, to enable this system and to decide whether players should be allowed to disable autobroadcast or not (can do this by setting min required level).

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:


1. Have CFBG OFF (important, they use their own queue announcer, needs to be updated)
2. Enable queue announcers (world.config file)
3. run .settings announcer bg off
4. queue to a bg, notice the announcer won't fire

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
